2025_rcw_hackday_tdastro_nb/model_from_prototype.ipynb

```diff
@@ -0,0 +1,1645 @@
+      "text/plain": [
+       "<Figure size 640x480 with 1 Axes>"
+      ]
+     },
+     "metadata": {},
+     "output_type": "display_data"
+    }
+   ],
+   "source": [
+    "mag_pred_grid = model.apply(params, t, yerr, y, t_gp_grid)[1]\n",
+    "pred_df = pd.DataFrame(\n",
+    "    {\n",
+    "        \"band\": band_gp_grid,\n",
+    "        \"mjd\": phase_gp_grid * PERIOD,\n",
+    "        \"mag\": mag_pred_grid,\n",
+    "        \"magerr\": np.zeros(N_BANDS * n_grid_band),\n",
+    "    }\n",
+    ")\n",
+    "plot_lc(pred_df, period=PERIOD)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "102b85ed392c25d3",
+   "metadata": {},
+   "source": [
+    "### 3. Simulate a single object\n",
+    "\n",
+    "Now when we have a function which gives us apparent magnitude of our source for any of LSST (and two of ZTF!) passbands, we can use it to create a \"light-curve source\".\n",
+    "Let's start with the built-in `LightcurveSource` class, and in the next section we will extend it to simulate a population."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "817d31a3f46d9f23",
+   "metadata": {},
+   "source": [
+    "#### Create `LightcurveSource`\n",
+    "\n",
+    "`LightcurveSource` class takes a \"grid\" model and interpolates it in time, with an optional support of periodic sources.\n",
+    "The grid model is basically \"perfect\" observations of our light curve, for all passbands.\n",
+    "\n",
+    "**Note**: tdastro, as well as Rubin, operates with bandfluxes in nJy, not with magnitudes.\n",
+    "\n",
+    "First of all, we will run our model on a grid, convert outputs to nJy, and reformat the result to use it with `LightcurveSource` constructor."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 11,
+   "id": "89b6ebb67c7c2e5d",
+   "metadata": {
+    "ExecuteTime": {
+     "end_time": "2025-07-25T17:26:10.963127Z",
+     "start_time": "2025-07-25T17:26:08.482436Z"
+    }
+   },
+   "outputs": [],
+   "source": [
+    "def bandflux_model(phase, band):\n",
+    "    \"\"\"Output bandflux in nJy for given MJD and band.\"\"\"\n",
+    "    orig_shape = shape = np.broadcast_shapes(np.shape(phase), np.shape(band))\n",
+    "    if shape == ():\n",
+    "        shape = (1,)\n",
+    "    phase = np.broadcast_to(phase, shape)\n",
+    "    band = np.broadcast_to(band, shape)\n",
+    "    band_one_hot = np.asarray([BAND_ENCODINGS[x] for x in band])\n",
+    "    t_grid = np.hstack([phase[:, None], band_one_hot])\n",
+    "    mag = np.asarray(model.apply(params, t, yerr, y, t_grid)[1])\n",
+    "    flux = mag2flux(mag)\n",
+    "    return flux.reshape(orig_shape)\n",
+    "\n",
+    "\n",
+    "n_source_grid = 1000\n",
```

Comment:
It might be worth adding a comment saying that you are creating a grid of times such that they cover one period.